src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs

```diff
@@ -125,18 +128,18 @@ private void OnResolve(Type serviceType, IServiceScope scope)
             _callSiteValidator?.ValidateResolution(serviceType, scope, Root);
         }
 
-        internal object? GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
+        internal object? GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
         {
             if (_disposed)
             {
                 ThrowHelper.ThrowObjectDisposedException();
             }
 
-            Func<ServiceProviderEngineScope, object?> realizedService = _realizedServices.GetOrAdd(serviceType, _createServiceAccessor);
-            OnResolve(serviceType, serviceProviderEngineScope);
-            DependencyInjectionEventSource.Log.ServiceResolved(this, serviceType);
+            Func<ServiceProviderEngineScope, object?> realizedService = _realizedServices.GetOrAdd(serviceIdentifier, _createServiceAccessor);
+            OnResolve(serviceIdentifier.ServiceType, serviceProviderEngineScope); // TODO NOT SURE
```

Comment:
Shouldn't we add the name to the ServiceCallSite? And to CallSiteValidator.ValidateResolution()? Right now, I don't think our CallSiteValidator is going to identity any scoped-service misusage for named services. We should add a test for this.

Reply:
It looks like the _createServiceAccessor holds the service name, but looks like on resolution there's just a service type for lookup.